### PR TITLE
Fix setImmediate pollyfill in strict mode

### PR DESCRIPTION
**Pull request by sbc100**
This required adding a browser test that includes this pollyfill
because the issue doesn't show up in node (setImmediate *is* defined
in node).

Fixes: #12568

### CODE REVIEW

File: src/library_html5.js [hunk 1]
```diff
@@ -2672,6 +2672,23 @@ var LibraryJSEvents = {
...
+    // nop, used to pulli in globalThis__postset
+  },
+
+  $globalThis__postset: `
+    (function() {
+      if (typeof globalThis === 'object') return;
+      Object.prototype.__defineGetter__('__magic__', function() {
+        return this;
+      });
+      __magic__.globalThis = __magic__; // lolwat
```
Comment (surma) [inline src/library_html5.js:2686]: @mathiasbynens Look!

**Review by surma — APPROVED** [review]
🎉

File: src/library_html5.js (hunk 1 above)
Comment (sbc100) [reply src/library_html5.js:2686]: I this a sensible thing to push in production.. i have no idea how this is working :)

Comment (hoodmane) [reply src/library_html5.js:2686]: https://mathiasbynens.be/notes/globalthis

**Review by kripken — COMMENTED** [review]
I'm missing something here. Why do we want to polyfill `setImmediate` globally? I'd think it's better to polyfill it "locally" as much as possible, that is, in a way that allows it to be limited to this module. I admit I don't know of a way to do that well enough: `if (typeof setImmediate === 'undefined') setImmediate = ...` still writes globally, and if you write `var setImmediate` then it doesn't pick up the existing code. But are we sure there isn't a good way, before writing to globalThis?

**Comment by sbc100** [comment]
How about this method which doesn't mess with the globalThis ?

**Review by kripken — APPROVED** [review]
Thanks, this looks better to me.

Yes, my concern is affecting outside code. We should try to avoid that as much as possible.

**Pull request merged**